src/eureka/S1_detector_processing/group_level.py

```diff
@@ -38,8 +38,17 @@ def GLBS(input_model, log, meta):
     dq = input_model.groupdq
 
     meta.inst = input_model.meta.instrument.name.lower()
-    meta.int_start = 0
     meta.n_int = all_data.shape[0]
+    meta.int_start = 0
+    if not hasattr(meta, 'nplots') or meta.nplots is None:
+        meta.int_end = meta.n_int
+    elif meta.int_start+meta.nplots > meta.n_int:
+        # Too many figures requested, so reduce it
+        meta.int_end = meta.n_int
+    else:
+        meta.int_end = meta.int_start+meta.nplots
+    if meta.inst == 'miri':
+        meta.isrotate = False
```

Comment:
Why not rotate the MIRI frames?

Reply:
`isrotate` is true for the NIR instruments, so it needs to be false for MIRI.  For some background info on the matter... I originally wrote `background.fitbg()` for an instrument that dispersed light along the y axis (like MIRI) and then modified it using the `isrotate` command for instruments that disperse light along x.